front/src/common/Map/MapKey.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { catenaryMapKey, MapKeyProps, speedLimitMapKey } from './const';
+import colors from 'common/Map/Consts/colors';
+import 'common/Map/MapKey.scss';
+
+const MapSettings = ({ active, toggleMapKey }: MapKeyProps) => {
+  const { t } = useTranslation(['translation', 'map-key']);
+  const { mapStyle } = useSelector((state: any) => state.map);
+
+  return (
+    active && (
+      <div className={`map-modal map-modal-dark active`}>
+        <div className="h2 text-light">{t('map-key:keyTitle')}</div>
+        <div className="row">
+          <div className="col-lg-6">
+            <div className="mapkey">
+              <div className="mapkey-title">{t('map-key:speedlimit')}</div>
+              {speedLimitMapKey.map((key) => {
+                return (
+                  <div className="mapkey-item" key={key.text}>
+                    <div className="mapkey-icon">
+                      <i className={`mapkey-line mapkey-${key.speed}`} />
+                    </div>
+                    <div className="mapkey-text">{key.text}</div>
+                  </div>
+                );
+              })}
+            </div>
+          </div>
+          <div className="col-lg-6">
+            <div className="mapkey">
+              <div className="mapkey-title">{t('map-key:catenaries')}</div>
+              {catenaryMapKey.map((key) => {
+                return (
+                  <div className="mapkey-item" key={key.text}>
+                    <div className="mapkey-icon">
+                      <i
+                        className="powerline"
+                        style={{
+                          background: `repeating-linear-gradient(to right, ${
+                            colors[mapStyle].powerline[key.color]
+                          } 20px, #333 22px, #333 24px)`,
+                        }}
+                      />
+                    </div>
+                    <div className="mapkey-text">
+                      {`${key.text} `}
+                      {key.current ? t(`map-key:${key.current}`) : null}
+                    </div>
+                  </div>
+                );
+              })}
+            </div>
+          </div>
+        </div>
+
+        <div className="mt-2 d-flex flex-row-reverse w-100">
+          <button className="btn btn-secondary btn-sm" type="button" onClick={() => toggleMapKey()}>
+            {t('translation:common.close')}
+          </button>
+        </div>
+      </div>
+    )
+  );
+};
+
+MapSettings.propTypes = {
```

Comment:
You can delete propTypes now that file is in Ts

Reply:
done